ibis/backends/decompile/compiler.py

```diff
@@ -0,0 +1,145 @@
+import functools
+import io
+import itertools
+
+import black
+
+import ibis.common.graph as graph
+import ibis.expr.datatypes as dt
+import ibis.expr.operations as ops
+import ibis.expr.schema as sch
+
+
+@functools.singledispatch
+def translate(op, *args, **kwargs):
+    raise NotImplementedError(op)
+
+
+@translate.register(dt.DataType)
+def datatype(dtype, *args, **kwargs):
+    return str(dtype)
+
+
+@translate.register(sch.Schema)
+def schema(schema, *, names, types):
+    return repr(list(zip(names, map(str, types))))
+
+
+@translate.register(ops.UnboundTable)
+def table(op, schema, name):
+    return f"ibis.table(name={name!r}, schema={schema})"
+
+
+@translate.register(ops.Selection)
+def selection(op, table, selections, predicates, sort_keys):
+    out = f"{table}"
+    if selections:
+        out = f"{out}.select({selections})"
+    if predicates:
+        out = f"{out}.filter({predicates})"
+    if sort_keys:
+        out = f"{out}.sort_by({sort_keys})"
+    return out
+
+
+@translate.register(ops.Limit)
+def limit(op, table, n, offset):
+    if offset:
+        return f"{table}.limit({n}, {offset})"
+    else:
+        return f"{table}.limit({n})"
+
+
+@translate.register(ops.TableColumn)
+def table_column(op, table, name):
+    return f"{table}.{name}"
+
+
+@translate.register(ops.SortKey)
+def sort_key(op, expr, ascending):
+    if ascending:
+        return f"ibis.asc({expr})"
+    else:
+        return f"ibis.desc({expr})"
+
+
+@translate.register(ops.Reduction)
+def reduction(op, arg, where, **kwargs):
+    name = op.__class__.__name__.lower()
+    return f"{arg}.{name}()"
+
+
+@translate.register(ops.Alias)
+def alias(op, arg, name):
+    return f"{arg}.name({name!r})"
+
+
+@translate.register(ops.Literal)
+def literal(op, value, dtype):
+    return repr(value)
+
+
+@translate.register(ops.NodeList)
+def node_list(op, *values):
+    return ", ".join(values)
+
+
+@translate.register(ops.Equals)
```

Comment:
Instead of defining each of the non-syntactic APIs, e.g., `StringColumn.split()` can we have a generic `Value` implementation that looks at a new class property and if that's `None` (the default) snake cases the operation class name to be the method call?

Otherwise it's a super annoying game of whack-a-mole for every operation.

Reply:
@cpcloud I may misunderstand, do you mean we should introduce a new class property?

Reply:
This is resolved!